runtime/sema/type.go

```diff
@@ -1621,6 +1626,44 @@ func getArrayMembers(arrayType ArrayType) map[string]MemberResolver {
 				)
 			},
 		},
+		"firstIndex": {
+			Kind: common.DeclarationKindFunction,
+			Resolve: func(identifier string, targetRange ast.Range, report func(error)) *Member {
+
+				elementType := arrayType.ElementType(false)
+
+				// It is impossible for an array of resources to have a `firstIndex` function:
+				// if the resource is passed as an argument, it cannot be inside the array
+
+				if elementType.IsResourceType() {
+					report(
+						&InvalidResourceArrayMemberError{
+							Name:            identifier,
+							DeclarationKind: common.DeclarationKindFunction,
+							Range:           targetRange,
+						},
+					)
+				}
+
+				// TODO: implement Equatable interface: https://github.com/dapperlabs/bamboo-node/issues/78
```

Comment:
Is this comment needed? Probably outdated now?

Reply:
Yes, the comment/TODO is still relevant, but the URL is outdated and should be removed:
```suggestion
				// TODO: implement Equatable interface
```